monitoring/prometheus.go

```diff
@@ -0,0 +1,125 @@
+package monitoring
+
+import (
+	"net/http"
+	"sync"
+
+	grpc_prometheus "github.com/grpc-ecosystem/go-grpc-prometheus"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var (
+	// metricGroups is a global variable of all registered metrics
+	// projected by the mutex below. All new MetricGroups should add
+	// themselves to this map within the init() method of their file.
+	metricGroups = make(map[string]metricGroupFactory)
+
+	// activeGroups is a global map of all active metric groups. This can
+	// be used by some of the "static' package level methods to look up the
+	// target metric group to export observations.
+	activeGroups = make(map[string]MetricGroup)
+
+	// metricsMtx is a global mutex that should be held when accessing the
+	// global maps.
+	metricsMtx sync.Mutex
+)
+
+// PrometheusExporter is a metric exporter that uses Prometheus directly. The
+// internal server will interact with this struct in order to export relevant
+// metrics.
+type PrometheusExporter struct {
+	config *PrometheusConfig
+}
+
+// Start registers all relevant metrics with the Prometheus library, then
+// launches the HTTP server that Prometheus will hit to scrape our metrics.
+func (p *PrometheusExporter) Start() error {
+	// If we're not active, then there's nothing more to do.
+	if !p.config.Active {
+		return nil
+	}
+
+	// Before we register the rpc servers, we'll ensure that the collector
+	// will export latency metrics for the histogram.
+	grpc_prometheus.EnableHandlingTimeHistogram()
```

Comment:
I think let's also put this behind a flag similar to `lnd`? It can end up adding a lot of metrics, which can end up bloating prom a bit.

Reply:
right, even with the counters we have quite a bit of series. With the new library we can disable them if need it in the future too.